.github/workflows/test.yml

```diff
@@ -0,0 +1,27 @@
+name: Unit tests CI
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'yarn'
+    - name: Install dependencies
```

Comment:
In case a PR has many commits, each commit will trigger the job. We should cache the result after installing the dependencies to reduce the billable time. It is spending 56 seconds here.

![image](https://user-images.githubusercontent.com/22078891/218694400-62532fd7-bba9-44b1-8287-2b4a2b9ff65a.png)

Reply:
The running time is reduced due to the yarn cache from `actions/setup-node@v3`
<img width="1097" alt="image" src="https://user-images.githubusercontent.com/44129390/218730519-545b5cc8-2fb1-46aa-b54d-d856c8402f1e.png">